### PR TITLE
fix: miscellaneous remaining bug bash fixes

### PR DESCRIPTION
**Pull request by hanlinc27**
## Brief description. What is this change? 

- #### [Wording consistency for buttons and header tabs](https://www.notion.so/uwblueprintexecs/UI-Wording-Consistency-6dbde61c00144b408e370e670c984e8d)
- #### [Landing page "Start Now" button alignment](https://www.notion.so/uwblueprintexecs/Button-UI-Fixes-1a096cfc7209472bac24bc2d6bbe460f)
- #### [Donation Process confirmation page date formatting](https://www.notion.so/uwblueprintexecs/Unformatted-date-confirmation-displayed-after-donation-created-5e878ee735134f22a63eea4aae5c0d77)
- #### [Remove close icon for email verification](https://www.notion.so/uwblueprintexecs/UI-Thank-you-page-after-email-verification-38740eba2bae42ec9ad11d8d43bbb330)

### CODE REVIEW

File: frontend/src/components/common/Header.tsx [hunk 1]
```diff
@@ -54,14 +54,25 @@ const Header = (): JSX.Element => {
         >
           <HamburgerIcon color="black.100" />
         </IconButton>
-        <Link as={ReactLink} to={Routes.LANDING_PAGE}>
+        {authenticatedUser?.role === Role.DONOR ? (
+          <>
```
Comment (hanlinc27) [inline frontend/src/components/common/Header.tsx:58]: make sure the cfkw logo isn't routable to Home when the admin is logged in

**Comment by carissa-tang** [comment]
re: [Stacy's comment](https://www.notion.so/Button-UI-Fixes-1a096cfc7209472bac24bc2d6bbe460f?d=3c52873b4c7743f086c85d7dd192b89b) - the "Start now" button can be removed (designs were updated)

File: frontend/src/components/common/Header.tsx [hunk 2]
```diff
@@ -79,7 +90,7 @@ const Header = (): JSX.Element => {
                     Home
                   </Link>
                   <Link as={ReactLink} to={Routes.DASHBOARD_PAGE}>
-                    Scheduled Donations
+                    My Scheduled Donations
                   </Link>
                 </>
               )}
```
Comment (carissa-tang) [inline frontend/src/components/common/Header.tsx:96]: switch order for `User Management` and `View Donations` on Desktop as well (line 99)

Comment (hanlinc27) [reply frontend/src/components/common/Header.tsx:96]: good catch!

**Review by akshay2000saxena — APPROVED** [review]

**Pull request merged**